src/org/rust/lang/core/resolve/scope/RustResolveScope.kt

```diff
@@ -1,13 +1,12 @@
 package org.rust.lang.core.resolve.scope
 
-import com.intellij.psi.PsiElement
 import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustDeclaringElement
 import org.rust.lang.core.psi.RustNamedElement
-import org.rust.lang.core.psi.RustPatVar
 import org.rust.lang.core.resolve.RustResolveEngine
 
 public interface RustResolveScope : RustCompositeElement {
-    fun listDeclarations(before: PsiElement): List<RustPatVar>
+    fun getDeclarations(): Collection<RustDeclaringElement>
```

Comment:
We should not rely on order reported by the node, and even more: we should not perform any trickery on that side since all semantic-extraction stuff is going to be cached in the nearest future.

Therefore `Collection` just saves us the place for the consequent maneuaver(-s) and hoping back and forth among `List`, `HashSet`, etc.